test/unit/azure_rm_test.rb

```diff
@@ -0,0 +1,30 @@
+require_relative '../test_plugin_helper'
+
+module ForemanAzureRm
+  class ForemanAzureRmTest < ActiveSupport::TestCase
+
+    setup do
+      @test_sdk = mock('test_sdk') #creates an empty object
+      ForemanAzureRm::AzureRm.any_instance.stubs(:sdk).returns(@test_sdk)
+    end
+
+    def new_azurerm_cr
+      FactoryBot.build(:azure_rm)
```

Comment:
You can move this to `def setup` and set its sdk to `@test_sdk` 
```suggestion
      @azure_cr = FactoryBot.build(:azure_rm)
      @azure_cr.stubs(:sdk).returns(@test_sdk)
```

Now you can work with `@azure_cr` and test its behaviour given mocked `sdk` that is controlled by you.